I tried the following, but it results in the tests not finding aiohttp_cors.

```patch
diff --git a/nixos/modules/services/misc/home-assistant.nix b/nixos/modules/services/misc/home-assistant.nix
index 1f2e13f3732..7d0a031be72 100644
--- a/nixos/modules/services/misc/home-assistant.nix
+++ b/nixos/modules/services/misc/home-assistant.nix
@@ -183,8 +183,14 @@ in {
     };
 
     package = mkOption {
-      default = pkgs.home-assistant;
-      defaultText = "pkgs.home-assistant";
+      default = pkgs.home-assistant.overridePythonAttrs (oldAttrs: {
+        doCheck = false;
+      });
+      defaultText = literalExample ''
+        pkgs.home-assistant.overridePythonAttrs (oldAttrs: {
+          doCheck = false;
+        }
+      '';
       type = types.package;
       example = literalExample ''
         pkgs.home-assistant.override {
```

```
hass # [    5.818279] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain logger took 0.0 seconds
hass # [    5.819942] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up http
hass # [    5.825566] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Error during setup of component http
hass # [    5.826350] hass[722]: Traceback (most recent call last):
hass # [    5.826836] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/setup.py", line 213, in _async_setup_component
hass # [    5.828039] hass[722]:     result = await task
hass # [    5.828470] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 210, in async_setup
hass # [    5.830821] hass[722]:     server = HomeAssistantHTTP(
hass # [    5.831271] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 316, in __init__
hass # [    5.832405] hass[722]:     setup_cors(app, cors_origins)
hass # [    5.833560] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/cors.py", line 26, in setup_cors
hass # [    5.834864] hass[722]:     import aiohttp_cors
hass # [    5.835553] hass[722]: ModuleNotFoundError: No module named 'aiohttp_cors'
hass # [    5.836764] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of system_log. Setup failed for dependencies: http
hass # [    5.838257] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for system_log: Could not set up all dependencies.
hass # [    5.839609] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 1: {'image', 'lovelace', 'device_automation', 'auth', 'person', 'search', 'system_log', 'onboarding', 'frontend', 'websocket_api', 'api', 'config', 'http'}
hass # [    5.867062] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up lovelace
hass # [    5.874013] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up device_automation
hass # [    5.874873] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain device_automation took 0.0 seconds
hass # [    5.876540] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of image. Setup failed for dependencies: http
hass # [    5.877624] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for image: Could not set up all dependencies.
hass # [    5.878799] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of auth. Setup failed for dependencies: http
hass # [    5.879682] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for auth: Could not set up all dependencies.
hass # [    5.881515] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of websocket_api. Setup failed for dependencies: http
hass # [    5.883960] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for websocket_api: Could not set up all dependencies.
hass # [    5.885806] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of api. Setup failed for dependencies: http
hass # [    5.886679] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for api: Could not set up all dependencies.
hass # [    5.888477] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of config. Setup failed for dependencies: http
hass # [    5.890927] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for config: Could not set up all dependencies.
hass # [    5.892819] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of person. Setup failed for dependencies: image
hass # [    5.894459] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for person: Could not set up all dependencies.
hass # [    5.895397] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of search. Setup failed for dependencies: websocket_api
hass # [    5.896604] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for search: Could not set up all dependencies.
hass # [    5.897987] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain lovelace took 0.0 seconds
hass # [    5.899008] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of onboarding. Setup failed for dependencies: auth, http, person
hass # [    5.900292] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for onboarding: Could not set up all dependencies.
hass # [    5.901780] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of frontend. Setup failed for dependencies: api, auth, config, http, onboarding, search, system_log, websocket_api
hass # [    5.903498] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for frontend: Could not set up all dependencies.
hass # [    5.904973] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'binary_sensor', 'mqtt'}
hass # [    5.957993] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up binary_sensor
hass # [    5.958908] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain binary_sensor took 0.0 seconds
hass # [    5.960115] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of mqtt. Setup failed for dependencies: http
hass # [    5.961802] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for mqtt: Could not set up all dependencies.
hass # [    5.963800] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of mqtt. Setup failed for dependencies: http
hass # [    5.964682] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform mqtt.binary_sensor: Could not set up all dependencies.
hass # [    5.966991] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 0.20s
hass # [    5.967995] hass[722]: 2021-03-03 23:58:00 WARNING (MainThread) [homeassistant.bootstrap] Detected that frontend did not load. Activating safe mode
hass # [    5.970798] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Starting in safe mode
hass # [    5.971548] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded homeassistant from homeassistant.components.homeassistant
hass # [    5.973184] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded persistent_notification from homeassistant.components.persistent_notification
hass # [    5.974796] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up homeassistant
hass # [    5.976160] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain homeassistant took 0.0 seconds
hass # [    5.978032] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up persistent_notification
hass # [    5.979673] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain persistent_notification took 0.0 seconds
hass # [    5.982186] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded http from homeassistant.components.http
hass # [    5.983808] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded safe_mode from homeassistant.components.safe_mode
hass # [    5.985548] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded frontend from homeassistant.components.frontend
hass # [    5.987764] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded api from homeassistant.components.api
hass # [    5.988971] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded auth from homeassistant.components.auth
hass # [    5.990722] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded config from homeassistant.components.config
hass # [    5.992177] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded device_automation from homeassistant.components.device_automation
hass # [    5.993794] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded lovelace from homeassistant.components.lovelace
hass # [    5.995001] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded onboarding from homeassistant.components.onboarding
hass # [    6.000162] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded person from homeassistant.components.person
hass # [    6.000976] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded image from homeassistant.components.image
hass # [    6.001750] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded search from homeassistant.components.search
hass # [    6.003573] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded websocket_api from homeassistant.components.websocket_api
hass # [    6.005409] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_0) [homeassistant.loader] Loaded system_log from homeassistant.components.system_log
hass # [    6.007899] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_3) [homeassistant.loader] Loaded cloud from homeassistant.components.cloud
hass # [    6.008720] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_1) [homeassistant.loader] Loaded webhook from homeassistant.components.webhook
hass # [    6.010566] hass[722]: 2021-03-03 23:58:00 INFO (SyncWorker_2) [homeassistant.loader] Loaded alexa from homeassistant.components.alexa
hass # [    6.012345] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Domains to be set up: {'image', 'cloud', 'device_automation', 'auth', 'lovelace', 'person', 'search', 'webhook', 'api', 'config', 'alexa', 'onboarding', 'frontend', 'websocket_api', 'system_log', 'safe_mode', 'persistent_notification', 'http'}
hass # [    6.015901] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up logging: {'system_log'}
hass # [    6.016635] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up http
hass # [    6.018337] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Error during setup of component http
hass # [    6.020910] hass[722]: Traceback (most recent call last):
hass # [    6.021378] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/setup.py", line 213, in _async_setup_component
hass # [    6.023949] hass[722]:     result = await task
hass # [    6.024384] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 210, in async_setup
hass # [    6.026205] hass[722]:     server = HomeAssistantHTTP(
hass # [    6.026791] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/__init__.py", line 316, in __init__
hass # [    6.028530] hass[722]:     setup_cors(app, cors_origins)
hass # [    6.029462] hass[722]:   File "/nix/store/gvzp61ynjg1c9dr2jba4ja2swlbmm90l-homeassistant-2021.3.0/lib/python3.8/site-packages/homeassistant/components/http/cors.py", line 26, in setup_cors
hass # [    6.031272] hass[722]:     import aiohttp_cors
hass # [    6.031698] hass[722]: ModuleNotFoundError: No module named 'aiohttp_cors'
hass # [    6.032261] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of system_log. Setup failed for dependencies: http
hass # [    6.033141] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for system_log: Could not set up all dependencies.
hass # [    6.033976] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 1: {'cloud', 'image', 'device_automation', 'auth', 'lovelace', 'webhook', 'person', 'search', 'system_log', 'alexa', 'onboarding', 'frontend', 'websocket_api', 'api', 'config', 'http'}
hass # [    6.035361] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up device_automation
hass # [    6.036074] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain device_automation took 0.0 seconds
hass # [    6.036845] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setting up lovelace
hass # [    6.037484] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.setup] Setup of domain lovelace took 0.0 seconds
hass # [    6.038243] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of image. Setup failed for dependencies: http
hass # [    6.039171] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for image: Could not set up all dependencies.
hass # [    6.039995] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of auth. Setup failed for dependencies: http
hass # [    6.040861] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for auth: Could not set up all dependencies.
hass # [    6.041664] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of webhook. Setup failed for dependencies: http
hass # [    6.042631] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for webhook: Could not set up all dependencies.
hass # [    6.043465] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of alexa. Setup failed for dependencies: http
hass # [    6.044353] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for alexa: Could not set up all dependencies.
hass # [    6.045184] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of websocket_api. Setup failed for dependencies: http
hass # [    6.046142] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for websocket_api: Could not set up all dependencies.
hass # [    6.047063] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of api. Setup failed for dependencies: http
hass # [    6.047959] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for api: Could not set up all dependencies.
hass # [    6.048780] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of config. Setup failed for dependencies: http
hass # [    6.049675] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for config: Could not set up all dependencies.
hass # [    6.050509] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of person. Setup failed for dependencies: image
hass # [    6.051456] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for person: Could not set up all dependencies.
hass # [    6.052311] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of cloud. Setup failed for dependencies: http, webhook, alexa
hass # [    6.053262] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for cloud: Could not set up all dependencies.
hass # [    6.054071] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of search. Setup failed for dependencies: websocket_api
hass # [    6.055005] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for search: Could not set up all dependencies.
hass # [    6.055954] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of onboarding. Setup failed for dependencies: auth, http, person
hass # [    6.056956] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for onboarding: Could not set up all dependencies.
hass # [    6.057795] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of frontend. Setup failed for dependencies: api, auth, config, http, onboarding, search, system_log, websocket_api
hass # [    6.058952] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for frontend: Could not set up all dependencies.
hass # [    6.059784] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'safe_mode', 'persistent_notification'}
hass # [    6.060690] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Unable to set up dependencies of safe_mode. Setup failed for dependencies: frontend, cloud
hass # [    6.061661] hass[722]: 2021-03-03 23:58:00 ERROR (MainThread) [homeassistant.setup] Setup failed for safe_mode: Could not set up all dependencies.
hass # [    6.062478] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 0.06s
hass # [    6.063211] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.core] Starting Home Assistant
hass # [    6.063888] hass[722]: 2021-03-03 23:58:00 INFO (MainThread) [homeassistant.core] Timer:starting
```

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -2,7 +2,7 @@
 # Do not edit!
 
 {
-  version = "2021.2.3";
+  version = "2021.3.0";
   components = {
     "abode" = ps: with ps; [ abodepy ];
     "accuweather" = ps: with ps; [ accuweather ];
@@ -12,6 +12,7 @@
     "adguard" = ps: with ps; [ adguardhome ];
     "ads" = ps: with ps; [ pyads ];
     "advantage_air" = ps: with ps; [ advantage-air ];
+    "aemet" = ps: with ps; [ ]; # missing inputs: AEMET-OpenData
     "aftership" = ps: with ps; [ pyaftership ];
     "agent_dvr" = ps: with ps; [ agent-py ];
     "air_quality" = ps: with ps; [ ];
@@ -120,13 +121,13 @@
     "clickatell" = ps: with ps; [ ];
     "clicksend" = ps: with ps; [ ];
     "clicksend_tts" = ps: with ps; [ ];
+    "climacell" = ps: with ps; [ ]; # missing inputs: pyclimacell
     "climate" = ps: with ps; [ ];
     "cloud" = ps: with ps; [ aiohttp-cors hass-nabucasa ];
     "cloudflare" = ps: with ps; [ pycfdns ];
     "cmus" = ps: with ps; [ ]; # missing inputs: pycmus
     "co2signal" = ps: with ps; [ ]; # missing inputs: co2signal
     "coinbase" = ps: with ps; [ ]; # missing inputs: coinbase
-    "coinmarketcap" = ps: with ps; [ coinmarketcap ];
     "color_extractor" = ps: with ps; [ ]; # missing inputs: colorthief
     "comed_hourly_pricing" = ps: with ps; [ ];
     "comfoconnect" = ps: with ps; [ ]; # missing inputs: pycomfoconnect
@@ -142,7 +143,6 @@
     "cover" = ps: with ps; [ ];
     "cppm_tracker" = ps: with ps; [ ]; # missing inputs: clearpasspy
     "cpuspeed" = ps: with ps; [ py-cpuinfo ];
-    "crimereports" = ps: with ps; [ ]; # missing inputs: crimereports
     "cups" = ps: with ps; [ pycups ];
     "currencylayer" = ps: with ps; [ ];
     "daikin" = ps: with ps; [ pydaikin ];
@@ -154,7 +154,7 @@
     "deconz" = ps: with ps; [ ]; # missing inputs: pydeconz
     "decora" = ps: with ps; [ bluepy ]; # missing inputs: decora
     "decora_wifi" = ps: with ps; [ ]; # missing inputs: decora_wifi
-    "default_config" = ps: with ps; [ pynacl aiohttp-cors defusedxml distro emoji hass-nabucasa netdisco pillow scapy sqlalchemy zeroconf ];
+    "default_config" = ps: with ps; [ pynacl aiohttp-cors async-upnp-client defusedxml distro emoji hass-nabucasa netdisco pillow scapy sqlalchemy zeroconf ];
     "delijn" = ps: with ps; [ ]; # missing inputs: pydelijn
     "deluge" = ps: with ps; [ deluge-client ];
     "demo" = ps: with ps; [ aiohttp-cors ];
@@ -238,6 +238,7 @@
     "everlights" = ps: with ps; [ pyeverlights ];
     "evohome" = ps: with ps; [ ]; # missing inputs: evohome-async
     "ezviz" = ps: with ps; [ ]; # missing inputs: pyezviz
+    "faa_delays" = ps: with ps; [ faadelays ];
     "facebook" = ps: with ps; [ ];
     "facebox" = ps: with ps; [ ];
     "fail2ban" = ps: with ps; [ ];
@@ -276,7 +277,7 @@
     "foscam" = ps: with ps; [ ]; # missing inputs: libpyfoscam
     "foursquare" = ps: with ps; [ aiohttp-cors ];
     "free_mobile" = ps: with ps; [ ]; # missing inputs: freesms
-    "freebox" = ps: with ps; [ aiohttp-cors netdisco zeroconf ]; # missing inputs: aiofreepybox
+    "freebox" = ps: with ps; [ aiohttp-cors netdisco zeroconf ]; # missing inputs: freebox-api
     "freedns" = ps: with ps; [ ];
     "fritz" = ps: with ps; [ fritzconnection ];
     "fritzbox" = ps: with ps; [ pyfritzhome ];
@@ -421,6 +422,7 @@
     "keyboard_remote" = ps: with ps; [ aionotify evdev ];
     "kira" = ps: with ps; [ pykira ];
     "kiwi" = ps: with ps; [ ]; # missing inputs: kiwiki-client
+    "kmtronic" = ps: with ps; [ pykmtronic ];
     "knx" = ps: with ps; [ xknx ];
     "kodi" = ps: with ps; [ ]; # missing inputs: pykodi
     "konnected" = ps: with ps; [ aiohttp-cors ]; # missing inputs: konnected
@@ -446,6 +448,7 @@
     "linux_battery" = ps: with ps; [ batinfo ];
     "lirc" = ps: with ps; [ ]; # missing inputs: python-lirc
     "litejet" = ps: with ps; [ ]; # missing inputs: pylitejet
+    "litterrobot" = ps: with ps; [ ]; # missing inputs: pylitterbot
     "llamalab_automate" = ps: with ps; [ ];
     "local_file" = ps: with ps; [ ];
     "local_ip" = ps: with ps; [ ];
@@ -466,6 +469,7 @@
     "lutron_caseta" = ps: with ps; [ ]; # missing inputs: aiolip pylutron-caseta
     "lw12wifi" = ps: with ps; [ ]; # missing inputs: lw12
     "lyft" = ps: with ps; [ ]; # missing inputs: lyft_rides
+    "lyric" = ps: with ps; [ aiohttp-cors aiolyric ];
     "magicseaweed" = ps: with ps; [ ]; # missing inputs: magicseaweed
     "mailbox" = ps: with ps; [ aiohttp-cors ];
     "mailgun" = ps: with ps; [ aiohttp-cors ]; # missing inputs: pymailgunner
@@ -476,6 +480,7 @@
     "mastodon" = ps: with ps; [ ]; # missing inputs: Mastodon.py
     "matrix" = ps: with ps; [ matrix-client ];
     "maxcube" = ps: with ps; [ ]; # missing inputs: maxcube-api
+    "mazda" = ps: with ps; [ pymazda ];
     "mcp23017" = ps: with ps; [ ]; # missing inputs: RPi.GPIO adafruit-circuitpython-mcp230xx
     "media_extractor" = ps: with ps; [ aiohttp-cors youtube-dl-light ];
     "media_player" = ps: with ps; [ aiohttp-cors ];
@@ -519,7 +524,9 @@
     "mqtt_room" = ps: with ps; [ aiohttp-cors paho-mqtt ];
     "mqtt_statestream" = ps: with ps; [ aiohttp-cors paho-mqtt ];
     "msteams" = ps: with ps; [ pymsteams ];
+    "mullvad" = ps: with ps; [ mullvad-api ];
     "mvglive" = ps: with ps; [ PyMVGLive ];
+    "my" = ps: with ps; [ aiohttp-cors pillow ];
     "mychevy" = ps: with ps; [ ]; # missing inputs: mychevy
     "mycroft" = ps: with ps; [ ]; # missing inputs: mycroftapi
     "myq" = ps: with ps; [ pymyq ];
@@ -560,7 +567,6 @@
     "nsw_fuel_station" = ps: with ps; [ ]; # missing inputs: nsw-fuel-api-client
     "nsw_rural_fire_service_feed" = ps: with ps; [ ]; # missing inputs: aio_geojson_nsw_rfs_incidents
     "nuheat" = ps: with ps; [ ]; # missing inputs: nuheat
-    "nuimo_controller" = ps: with ps; [ ]; # missing inputs: --only-binary=all nuimo
     "nuki" = ps: with ps; [ pynuki ];
     "numato" = ps: with ps; [ ]; # missing inputs: numato-gpio
     "number" = ps: with ps; [ ];
@@ -622,7 +628,7 @@
     "ping" = ps: with ps; [ icmplib ];
     "pioneer" = ps: with ps; [ ];
     "pjlink" = ps: with ps; [ ]; # missing inputs: pypjlink2
-    "plaato" = ps: with ps; [ aiohttp-cors ];
+    "plaato" = ps: with ps; [ aiohttp-cors hass-nabucasa pyplaato ];
     "plant" = ps: with ps; [ sqlalchemy ];
     "plex" = ps: with ps; [ aiohttp-cors plexapi plexauth plexwebsocket ];
     "plugwise" = ps: with ps; [ plugwise ];
@@ -682,6 +688,7 @@
     "ring" = ps: with ps; [ ha-ffmpeg ring-doorbell ];
     "ripple" = ps: with ps; [ ]; # missing inputs: python-ripple-api
     "risco" = ps: with ps; [ pyrisco ];
+    "rituals_perfume_genie" = ps: with ps; [ pyrituals ];
     "rmvtransport" = ps: with ps; [ PyRMVtransport ];
     "rocketchat" = ps: with ps; [ ]; # missing inputs: rocketchat-API
     "roku" = ps: with ps; [ ]; # missing inputs: rokuecp
@@ -749,6 +756,7 @@
     "smart_meter_texas" = ps: with ps; [ ]; # missing inputs: smart-meter-texas
     "smarthab" = ps: with ps; [ ]; # missing inputs: smarthab
     "smartthings" = ps: with ps; [ aiohttp-cors hass-nabucasa ]; # missing inputs: pysmartapp pysmartthings
+    "smarttub" = ps: with ps; [ python-smarttub ];
     "smarty" = ps: with ps; [ ]; # missing inputs: pysmarty
     "smhi" = ps: with ps; [ ]; # missing inputs: smhi-pkg
     "sms" = ps: with ps; [ python-gammu ];
@@ -780,7 +788,7 @@
     "sql" = ps: with ps; [ sqlalchemy ];
     "squeezebox" = ps: with ps; [ pysqueezebox ];
     "srp_energy" = ps: with ps; [ ]; # missing inputs: srpenergy
-    "ssdp" = ps: with ps; [ aiohttp-cors defusedxml netdisco zeroconf ];
+    "ssdp" = ps: with ps; [ aiohttp-cors async-upnp-client defusedxml netdisco zeroconf ];
     "starline" = ps: with ps; [ ]; # missing inputs: starline
     "starlingbank" = ps: with ps; [ ]; # missing inputs: starlingbank
     "startca" = ps: with ps; [ xmltodict ];
@@ -792,6 +800,7 @@
     "stream" = ps: with ps; [ aiohttp-cors av ];
     "streamlabswater" = ps: with ps; [ ]; # missing inputs: streamlabswater
     "stt" = ps: with ps; [ aiohttp-cors ];
+    "subaru" = ps: with ps; [ subarulink ];
     "suez_water" = ps: with ps; [ ]; # missing inputs: pysuez
     "sun" = ps: with ps; [ ];
     "supervisord" = ps: with ps; [ ];
@@ -805,7 +814,6 @@
     "switcher_kis" = ps: with ps; [ aioswitcher ];
     "switchmate" = ps: with ps; [ ]; # missing inputs: pySwitchmate
     "syncthru" = ps: with ps; [ url-normalize ]; # missing inputs: pysyncthru
-    "synology" = ps: with ps; [ ]; # missing inputs: py-synology
     "synology_chat" = ps: with ps; [ ];
     "synology_dsm" = ps: with ps; [ ]; # missing inputs: synologydsm-api
     "synology_srm" = ps: with ps; [ ]; # missing inputs: synology-srm
@@ -873,7 +881,7 @@
     "twinkly" = ps: with ps; [ twinkly-client ];
     "twitch" = ps: with ps; [ python-twitch-client ];
     "twitter" = ps: with ps; [ twitterapi ];
-    "ubus" = ps: with ps; [ ];
+    "ubus" = ps: with ps; [ openwrt-ubus-rpc ];
     "ue_smart_radio" = ps: with ps; [ ];
     "uk_transport" = ps: with ps; [ ];
     "unifi" = ps: with ps; [ aiounifi ];
@@ -946,7 +954,6 @@
     "xbox" = ps: with ps; [ aiohttp-cors ]; # missing inputs: xbox-webapi
     "xbox_live" = ps: with ps; [ xboxapi ];
     "xeoma" = ps: with ps; [ pyxeoma ];
-    "xfinity" = ps: with ps; [ ]; # missing inputs: xfinity-gateway
     "xiaomi" = ps: with ps; [ ha-ffmpeg ];
     "xiaomi_aqara" = ps: with ps; [ pyxiaomigateway aiohttp-cors netdisco zeroconf ];
     "xiaomi_miio" = ps: with ps; [ construct python-miio ];

--- a/pkgs/servers/home-assistant/default.nix
+++ b/pkgs/servers/home-assistant/default.nix
@@ -136,6 +136,7 @@ in with py.pkgs; buildPythonApplication rec {
     # test infrastructure
     asynctest
     pytest-aiohttp
+    pytest-rerunfailures
     pytest-xdist
     pytestCheckHook
     requests-mock
@@ -292,7 +293,10 @@ in with py.pkgs; buildPythonApplication rec {
 
   pytestFlagsArray = [
     # limit amout of runners to reduce race conditions
-    "-n 2"
+    "-n auto"
+    # retry racy tests that end in "RuntimeError: Event loop is closed"
+    "--reruns 3"
+    "--only-rerun RuntimeError"
     # assign tests grouped by file to workers
     "--dist loadfile"
     # tests are located in tests/

--- a/pkgs/servers/home-assistant/default.nix
+++ b/pkgs/servers/home-assistant/default.nix
@@ -127,6 +127,8 @@ in with py.pkgs; buildPythonApplication rec {
     yarl
   ] ++ componentBuildInputs ++ extraBuildInputs;
 
+  makeWrapperArgs = lib.optional skipPip "--add-flags --skip-pip";
+
   # upstream only tests on Linux, so do we.
   doCheck = stdenv.isLinux;
 
@@ -139,49 +141,151 @@ in with py.pkgs; buildPythonApplication rec {
     requests-mock
     # component dependencies
     pyotp
+    respx
   ] ++ lib.concatMap (component: getPackages component py.pkgs) componentTests;
 
-  # We cannot test all components, since they'd introduce lots of dependencies, some of which are unpackaged,
-  # but we should test very common stuff, like what's in `default_config`.
-  # https://github.com/home-assistant/core/commits/dev/homeassistant/components/default_config/manifest.json
+  # We can reasonably test components that don't communicate with any network
+  # services. Before adding new components to this list make sure we have all
+  # its dependencies packaged and listed in ./component-packages.nix.
   componentTests = [
+    "alert"
     "api"
+    "auth"
     "automation"
+    "bayesian"
+    "binary_sensor"
+    "caldav"
+    "calendar"
+    "camera"
+    "climate"
+    "cloud"
+    "command_line"
     "config"
     "configurator"
+    "conversation"
     "counter"
+    "cover"
     "default_config"
     "demo"
+    "derivative"
+    "device_automation"
+    "device_sun_light_trigger"
+    "device_tracker"
     "dhcp"
     "discovery"
+    "emulated_hue"
+    "esphome"
+    "fan"
+    "ffmpeg"
+    "file"
+    "filesize"
+    "filter"
+    "flux"
+    "folder"
+    "folder_watcher"
+    "fritzbox"
+    "fritzbox_callmonitor"
     "frontend"
+    "generic"
+    "generic_thermostat"
+    "geo_json_events"
+    "geo_location"
     "group"
+    "hddtemp"
     "history"
+    "history_stats"
     "homeassistant"
+    "html5"
     "http"
     "hue"
+    "ifttt"
+    "image"
+    "image_processing"
+    "influxdb"
     "input_boolean"
     "input_datetime"
     "input_text"
     "input_number"
     "input_select"
+    "intent"
+    "intent_script"
+    "ipp"
+    "light"
+    "local_file"
+    "local_ip"
+    "lock"
     "logbook"
+    "logentries"
     "logger"
+    "lovelace"
+    "manual"
+    "manual_mqtt"
+    "media_player"
     "media_source"
+    "met"
     "mobile_app"
+    "modbus"
+    "moon"
+    "mqtt"
+    "mqtt_eventstream"
+    "mqtt_json"
+    "mqtt_room"
+    "mqtt_statestream"
+    "notify"
+    "number"
+    "ozw"
+    "panel_custom"
+    "panel_iframe"
+    "persistent_notification"
     "person"
+    "prometheus"
+    "proximity"
+    "push"
+    "python_script"
+    "random"
+    "recorder"
+    "rest"
+    "rest_command"
+    "rmvtransport"
+    "rss_feed_template"
+    "safe_mode"
     "scene"
     "script"
+    "search"
     "shell_command"
+    "shopping_list"
+    "simulated"
+    "sensor"
+    "smtp"
+    "sql"
     "ssdp"
+    "stream"
     "sun"
+    "switch"
     "system_health"
     "system_log"
     "tag"
+    "tasmota"
+    "tcp"
+    "template"
+    "threshold"
+    "time_date"
     "timer"
+    "tod"
+    "tts"
+    "universal"
+    "updater"
+    "upnp"
+    "uptime"
+    "vacuum"
+    "weather"
     "webhook"
     "websocket_api"
+    "wled"
+    "workday"
+    "worldclock"
     "zeroconf"
+    "zha"
     "zone"
     "zwave"
   ];
@@ -191,12 +295,17 @@ in with py.pkgs; buildPythonApplication rec {
     "-n 2"
     # assign tests grouped by file to workers
     "--dist loadfile"
-    # don't bulk test all components
-    "--ignore tests/components"
-    # pyotp since v2.4.0 complains about the short mock keys, hass pins v2.3.0
-    "--ignore tests/auth/mfa_modules/test_notify.py"
+    # tests are located in tests/
     "tests"
+    # dynamically add packages required for component tests
   ] ++ map (component: "tests/components/" + component) componentTests;
+
+  disabledTestPaths = [
+    # don't bulk test all components
+    "tests/components"
+    # pyotp since v2.4.0 complains about the short mock keys, hass pins v2.3.0
+    "tests/auth/mfa_modules/test_notify.py"
+  ];
 
   disabledTests = [
     # AssertionError: assert 1 == 0
@@ -212,8 +321,6 @@ in with py.pkgs; buildPythonApplication rec {
     # the tests require the existance of a media dir
     mkdir /build/media
   '';
-
-  makeWrapperArgs = lib.optional skipPip "--add-flags --skip-pip";
 
   passthru = {
     inherit (py.pkgs) hass-frontend;

--- a/pkgs/servers/home-assistant/default.nix
+++ b/pkgs/servers/home-assistant/default.nix
@@ -61,14 +61,14 @@ let
   extraBuildInputs = extraPackages py.pkgs;
 
   # Don't forget to run parse-requirements.py after updating
-  hassVersion = "2021.2.3";
+  hassVersion = "2021.3.0";
 
 in with py.pkgs; buildPythonApplication rec {
   pname = "homeassistant";
   version = assert (componentPackages.version == hassVersion); hassVersion;
 
   # check REQUIRED_PYTHON_VER in homeassistant/const.py
-  disabled = pythonOlder "3.7.1";
+  disabled = pythonOlder "3.8";
 
   # don't try and fail to strip 6600+ python files, it takes minutes!
   dontStrip = true;
@@ -80,7 +80,7 @@ in with py.pkgs; buildPythonApplication rec {
     owner = "home-assistant";
     repo = "core";
     rev = version;
-    sha256 = "0s1jcd94wwvmvzq86w8s9dwfvnmjs9l661z9pc6kwgagggjjgd8c";
+    sha256 = "19hdsnzrkkw1fihccmfjsgcg1m7l22a19sbmlh2dy272ckbnhj9r";
   };
 
   # leave this in, so users don't have to constantly update their downstream patch handling
@@ -88,15 +88,15 @@ in with py.pkgs; buildPythonApplication rec {
 
   postPatch = ''
     substituteInPlace setup.py \
+      --replace "aiohttp==3.7.4" "aiohttp>=3.7.3" \
       --replace "attrs==19.3.0" "attrs>=19.3.0" \
       --replace "bcrypt==3.1.7" "bcrypt>=3.1.7" \
-      --replace "awesomeversion==21.2.2" "awesomeversion>=21.2.2" \
-      --replace "cryptography==3.2" "cryptography" \
+      --replace "cryptography==3.3.2" "cryptography" \
       --replace "httpx==0.16.1" "httpx>=0.16.1" \
+      --replace "jinja2>=2.11.3" "jinja2>=2.11.2" \
       --replace "pip>=8.0.3,<20.3" "pip" \
-      --replace "pytz>=2020.5" "pytz>=2020.4" \
+      --replace "pytz>=2021.1" "pytz>=2020.5" \
       --replace "pyyaml==5.4.1" "pyyaml" \
-      --replace "requests==2.25.1" "requests>=2.25.0" \
       --replace "ruamel.yaml==0.15.100" "ruamel.yaml>=0.15.100"
     substituteInPlace tests/test_config.py --replace '"/usr"' '"/build/media"'
   '';
@@ -177,6 +177,7 @@ in with py.pkgs; buildPythonApplication rec {
     "emulated_hue"
     "esphome"
     "fan"
+    "faa_delays"
     "ffmpeg"
     "file"
     "filesize"
@@ -195,6 +196,7 @@ in with py.pkgs; buildPythonApplication rec {
     "hddtemp"
     "history"
     "history_stats"
+    "homekit_controller"
     "homeassistant"
     "html5"
     "http"
@@ -211,6 +213,7 @@ in with py.pkgs; buildPythonApplication rec {
     "intent"
     "intent_script"
     "ipp"
+    "kmtronic"
     "light"
     "local_file"
     "local_ip"
@@ -221,6 +224,7 @@ in with py.pkgs; buildPythonApplication rec {
     "lovelace"
     "manual"
     "manual_mqtt"
+    "mazda"
     "media_player"
     "media_source"
     "met"
@@ -232,6 +236,7 @@ in with py.pkgs; buildPythonApplication rec {
     "mqtt_json"
     "mqtt_room"
     "mqtt_statestream"
+    "mullvad"
     "notify"
     "number"
     "ozw"
@@ -239,6 +244,7 @@ in with py.pkgs; buildPythonApplication rec {
     "panel_iframe"
     "persistent_notification"
     "person"
+    "plaato"
     "prometheus"
     "proximity"
     "push"
@@ -247,6 +253,7 @@ in with py.pkgs; buildPythonApplication rec {
     "recorder"
     "rest"
     "rest_command"
+    "rituals_perfume_genie"
     "rmvtransport"
     "rss_feed_template"
     "safe_mode"
@@ -257,6 +264,7 @@ in with py.pkgs; buildPythonApplication rec {
     "shopping_list"
     "simulated"
     "sensor"
+    "smarttub"
     "smtp"
     "sql"
     "ssdp"
@@ -313,12 +321,16 @@ in with py.pkgs; buildPythonApplication rec {
 
   disabledTests = [
     # AssertionError: assert 1 == 0
+    "test_error_posted_as_event"
     "test_merge"
     # ModuleNotFoundError: No module named 'pyqwikswitch'
     "test_merge_id_schema"
     # keyring.errors.NoKeyringError: No recommended backend was available.
     "test_secrets_from_unrelated_fails"
     "test_secrets_credstash"
+    # generic/test_camera.py: AssertionError: 500 == 200
+    "test_fetching_without_verify_ssl"
+    "test_fetching_url_with_verify_ssl"
   ];
 
   preCheck = ''

--- a/pkgs/servers/home-assistant/frontend.nix
+++ b/pkgs/servers/home-assistant/frontend.nix
@@ -4,11 +4,11 @@ buildPythonPackage rec {
   # the frontend version corresponding to a specific home-assistant version can be found here
   # https://github.com/home-assistant/home-assistant/blob/master/homeassistant/components/frontend/manifest.json
   pname = "home-assistant-frontend";
-  version = "20210127.7";
+  version = "20210302.3";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-N+n1DI1oZ/j+GydH+Du21zTNA6leW4YBri7cAMvAac4=";
+    sha256 = "sha256-VJVYGg3xJC6npxYWw2h0Rb4jaeriOksN1NvAWVQzK50=";
   };
 
   # there is nothing to strip in this package

